src/nbl/asset/interchange/CGLTFLoader.h

```diff
@@ -0,0 +1,539 @@
+				}
+
+				_NBL_STATIC_INLINE std::string getSamplerCacheKey(const asset::ICPUSampler::SParams& params)
+				{
+					const std::size_t hash = std::hash<std::string_view>{}(std::string_view(reinterpret_cast<const char*>(&params), sizeof(params)));
+					return "nbl/builtin/samplers/loaders/glTF/" + std::to_string(hash);
+				}
+
+			private:
+
+				virtual void initialize() override;
+
+				using SSamplerCacheKey = std::string;
+				using SShaderDefinesCode = std::string;
+				using STextures = core::vector<std::pair<core::smart_refctd_ptr<ICPUImageView>, SSamplerCacheKey>>;
+
+				struct CGLTFHeader
+				{
+					uint32_t version;
+					std::optional<uint32_t> minVersion;
+					std::optional<std::string> generator;
+					std::optional<std::string> copyright;
+				};
+
+				/*
+					A node in the node hierarchy. A node can have either a matrix or any combination of 
+					translation/rotation/scale (TRS) properties.
+				*/
+
+				struct SGLTFNode
+				{
+					SGLTFNode() {}
+					virtual ~SGLTFNode() {}
+
+					std::optional<std::string> name;				//!< The user-defined name of this object.
+					std::optional<std::vector<uint32_t>> children;	//!< The indices of this node's children.
+					std::optional<uint32_t> mesh;					//!< The index of the mesh in this node.
+					std::optional<uint32_t> camera;					//!< The index of the camera referenced by this node.
+					std::optional<uint32_t> skin;					//!< The index of the skin referenced by this node.
+					std::optional<uint32_t> weights;				//!< The weights of the instantiated Morph Target. Number of elements must match number of Morph Targets of used mesh.
+
+					struct SGLTFNTransformationTRS
+					{
+						core::vector3df_SIMD translation;			//!< The node's translation along the x, y, and z axes.
+						core::vector3df_SIMD scale;					//!< The node's non-uniform scale, given as the scaling factors along the x, y, and z axes.
+						core::vector4df_SIMD rotation;				//!< The node's unit quaternion rotation in the order (x, y, z, w), where w is the scalar.
+					};
+
+					union SGLTFTransformation
+					{
+						SGLTFTransformation() {}
+						~SGLTFTransformation() {}
+
+						SGLTFTransformation(SGLTFTransformation& copy)
+						{
+							std::memmove(this, &copy, sizeof(SGLTFTransformation));
+						}
+
+						SGLTFTransformation(const SGLTFTransformation& copy)
+						{
+							std::memmove(this, &copy, sizeof(SGLTFTransformation));
+						}
+
+						SGLTFTransformation& operator=(SGLTFTransformation& copy)
+						{
+							std::memmove(this, &copy, sizeof(SGLTFTransformation));
+							return *this;
+						}
+
+						SGLTFTransformation& operator=(const SGLTFTransformation& copy)
+						{
+							std::memmove(this, &copy, sizeof(SGLTFTransformation));
+							return *this;
+						}
+
+						SGLTFNTransformationTRS trs;	
+						core::matrix4SIMD matrix;		//!< A floating-point 4x4 transformation matrix stored in column-major order.
```

Comment:
you can always convert a TRS to a matrix3x4SIMD, so transformation should always be kept as a matrix3x4SIMD

Reply:
corrected in following commit